graphql/src/main/java/com/linecorp/armeria/server/graphql/GraphQLServiceBuilder.java

```diff
@@ -0,0 +1,218 @@
+        runtimeWiringConfiguratorBuilder.addAll(requireNonNull(configurators, "configurators"));
+        return this;
+    }
+
+    /**
+     * Sets the {@link GraphQLTypeVisitor}.
+     */
+    public GraphQLServiceBuilder typeVisitors(GraphQLTypeVisitor... typeVisitors) {
+        return typeVisitors(ImmutableList.copyOf(requireNonNull(typeVisitors, "typeVisitors")));
+    }
+
+    /**
+     * Sets the {@link GraphQLTypeVisitor}.
+     */
+    public GraphQLServiceBuilder typeVisitors(Iterable<GraphQLTypeVisitor> typeVisitors) {
+        typeVisitorBuilder.addAll(requireNonNull(typeVisitors, "typeVisitors"));
+        return this;
+    }
+
+    /**
+     * Sets the {@link Instrumentation}.
+     */
+    public GraphQLServiceBuilder instrumentation(Instrumentation... instrumentations) {
+        return instrumentation(ImmutableList.copyOf(requireNonNull(instrumentations, "instrumentations")));
+    }
+
+    /**
+     * Sets the {@link Instrumentation}.
+     */
+    public GraphQLServiceBuilder instrumentation(Iterable<Instrumentation> instrumentations) {
+        instrumentationBuilder.addAll(requireNonNull(instrumentations, "instrumentations"));
+        return this;
+    }
+
+    /**
+     * Sets the {@link GraphQL.Builder} consumer.
+     */
+    public GraphQLServiceBuilder configure(Consumer<GraphQL.Builder>... configurers) {
+        return configure(ImmutableList.copyOf(requireNonNull(configurers, "configurers")));
+    }
+
+    /**
+     * Sets the {@link GraphQL.Builder} consumer.
+     */
+    public GraphQLServiceBuilder configure(Iterable<Consumer<GraphQL.Builder>> configurers) {
+        configurerBuilder.addAll(requireNonNull(configurers, "configurers"));
+        return this;
+    }
+
+    /**
+     * Creates a {@link GraphQLService}.
+     */
+    public GraphQLService build() {
+        final TypeDefinitionRegistry registry = typeDefinitionRegistry(schemaFileBuilder.build());
+
+        final RuntimeWiring runtimeWiring = buildRuntimeWiring(runtimeWiringConfiguratorBuilder.build());
+        GraphQLSchema schema = new SchemaGenerator().makeExecutableSchema(registry, runtimeWiring);
+        final List<GraphQLTypeVisitor> typeVisitors = typeVisitorBuilder.build();
+        for (GraphQLTypeVisitor typeVisitor : typeVisitors) {
+            schema = SchemaTransformer.transformSchema(schema, typeVisitor);
+        }
+
+        GraphQL.Builder builder = GraphQL.newGraphQL(schema);
+        final List<Instrumentation> instrumentations = instrumentationBuilder.build();
+        if (!instrumentations.isEmpty()) {
+            builder = builder.instrumentation(new ChainedInstrumentation(instrumentations));
+        }
+
+        final List<Consumer<GraphQL.Builder>> configurers = configurerBuilder.build();
+        for (Consumer<GraphQL.Builder> configurer : configurers) {
+            configurer.accept(builder);
+        }
+        return new DefaultGraphQLService(builder.build(), dataLoaderRegistry);
```

Comment:
Sometimes, users may want to directly build `GraphQLService` with an existing `GraphQL` instance.
Could we add `GraphQLService.of(GraphQL)`?